spec/workflow_service_spec.rb

```diff
@@ -27,56 +27,48 @@
   before(:each) do
     @repo  = 'dor'
     @druid = 'druid:123'
-    @mock_resource = double('mock_rest_client_resource')
-    @mock_http_connection = double('mock_http_connection')
-    @mock_logger = double('Logger')
-
-    allow(@mock_logger).to receive(:info)  # silence log output
-    allow(@mock_logger).to receive(:debug) # silence log output
-    allow(Dor::WorkflowService).to receive(:default_logger).and_return(@mock_logger)
-
+    @mock_resource = RestClient::Resource.new('https://localhost/workflow')
+    @mock_logger   = double('Logger')
     allow(@mock_resource).to receive(:[]).and_return(@mock_resource)
     allow(@mock_resource).to receive(:options).and_return( {} )
-    allow(@mock_resource).to receive(:url).and_return( 'https://localhost/workflow' )
-
+    allow(@mock_logger).to receive(:info)
+    allow(@mock_logger).to receive(:warn)
+    allow(@mock_logger).to receive(:debug) # silence logging output
+    @mock_http_connection = double('mock_http_connection')
     allow(Faraday).to receive(:new).and_return(@mock_http_connection)
-
-    allow(RestClient::Resource).to receive(:new).and_return(@mock_resource)
-    Dor::WorkflowService.configure @mock_resource.url
+    @dwfs = Dor::WorkflowService.new(@mock_resource, :logger => @mock_logger)
```

Comment:
Most of the changes in the specs are a global replace of `Dor::WorkflowService` for `@dwfs`.  Consider using `let` instead of instance vars and whether the `@dwfs` could be `subject`.